src/app/modules/user/user.controller.ts

```diff
@@ -57,12 +72,39 @@ export const handleContactVerifyOtp: RequestHandler<
   }
 
   // No error, update user with given contact.
-  const [updateErr] = await to(updateUserContact(contact, userId))
+  const [updateErr, updatedUser] = await to(updateUserContact(contact, userId))
   if (updateErr) {
     // Handle update error.
     logger.warn(updateErr)
     return res.status(HttpStatus.INTERNAL_SERVER_ERROR).send(updateErr.message)
   }
 
-  return res.sendStatus(HttpStatus.OK)
+  return res.status(HttpStatus.OK).send(updatedUser)
+}
+
+export const handleFetchUser: RequestHandler = async (req, res) => {
+  const sessionUserId = getUserIdFromSession(req.session)
+  if (!sessionUserId) {
+    return res.status(HttpStatus.UNAUTHORIZED).send('User is unauthorized.')
+  }
+
+  // Retrieve user with id in session
+  const [dbErr, retrievedUser] = await to(getPopulatedUserById(sessionUserId))
+
+  if (dbErr || !retrievedUser) {
+    logger.warn(
+      `handleFetchUser: Unable to retrieve user ${sessionUserId}`,
+      dbErr,
+    )
+    return res
+      .status(HttpStatus.INTERNAL_SERVER_ERROR)
+      .send('Unable to retrieve user')
+  }
+
+  return res.send(retrievedUser)
+}
+
+// TODO: Save userId instead of entire user collection in session.
```

Comment:
can make this TODO more specific? maybe open an issue? too many TODOs littered throughout the codebase which no one will ever follow up on